File: Labs/golang/Lab_12/Lab_12.go
package main

import (
	"fmt"
)

func main() {

	var current, oneYear, twoYear float64

	inputFunc(&current, &oneYear, &twoYear)
	result1 := calculate(current, oneYear)
	result2 := calculate(oneYear, twoYear)
	output(result1, result2)

}

func inputFunc(current, oneYear, twoYear *float64) {
	fmt.Println("Enter the current price")
	fmt.Scanln(current)

	fmt.Println("Enter last year's price")
	fmt.Scanln(oneYear)

	fmt.Println("Enter the price two years ago")
	fmt.Scanln(twoYear)
}

func calculate(current, past float64) float64 {
	return (current - past) / past
}

func output(result1, result2 float64) {
	if result1 > result2 {
		fmt.Println("Inflation is increasing")
		return
	}

	fmt.Println("Inflation is decreasing")
}

File: Labs/C++/Lab_12/Lab_12.cpp
// Lab 12
// Derek P Sifford
#include <iostream>
using namespace std;

void inputFunc(double& current, double& oneYear, double& twoYear) {

    cout << "Enter the current price" << endl;
    cin >> current;

    cout << "Enter the price one year ago" << endl;
    cin >> oneYear;

    cout << "Enter the price two years ago" << endl;
    cin >> twoYear;

}

double calculate(double current, double past) {
    return (current - past) / past;
}

void output(double result1, double result2) {
    if (result1 > result2) {
        cout << "Inflation is increasing" << endl;
        return;
    }

    if (result2 > result1) {
        cout << "Inflation is decreasing" << endl;
        return;
    }
}

int main() {

    double current, oneYear, twoYear, result1, result2;
    inputFunc(current, oneYear, twoYear);
    result1 = calculate(current, oneYear);
    result2 = calculate(oneYear, twoYear);
    output(result1, result2);

    return 0;
}
